### PR TITLE
[INLONG-6331][Dataproxy] To fix dataproxy prometheus problem

### LINKED ISSUE #6331: [Bug]Dataproxy is unable to obtain prometheus metrics correctly.
### What happened

When visit localhost:9081/metrics in agent, we will get empty reply from server.

### What you expected to happen

dataproxy get prometheus metrics correctly in localhost:9081/metrics.

### How to reproduce

run a agent job

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Keylchen**
### Prepare a Pull Request

- Fixes #6331

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java [hunk 1]
```diff
@@ -179,31 +175,33 @@ public void snapshot(String domain, List<MetricItemValue> itemValues) {
...
         totalCounter.addMetric(Arrays.asList(M_WHOLE_DURATION), metricItem.wholeDuration.get());
         List<MetricFamilySamples> mfs = new ArrayList<>();
         mfs.add(totalCounter);
 
         // id dimension
-        CounterMetricFamily idCounter = new CounterMetricFamily(metricName + "&group=id",
-                "The metrics of inlong datastream.", this.dimensionKeys);
+        List<String> dimensionIdKeys = new ArrayList<>();
+        dimensionIdKeys.add(DEFAULT_DIMENSION_LABEL);
+        dimensionIdKeys.addAll(this.dimensionKeys);
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java:202]: this.dimensionKeys is always empty.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java [hunk 2]
```diff
@@ -179,31 +175,33 @@ public void snapshot(String domain, List<MetricItemValue> itemValues) {
     public List<MetricFamilySamples> collect() {
 
         // total
-        CounterMetricFamily totalCounter = new CounterMetricFamily(metricName + "&group=total",
+        CounterMetricFamily totalCounter = new CounterMetricFamily(metricName,
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java:178]: Is it OK to change metric name?

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java [hunk 3]
```diff
@@ -179,31 +175,33 @@ public void snapshot(String domain, List<MetricItemValue> itemValues) {
...
         List<MetricFamilySamples> mfs = new ArrayList<>();
         mfs.add(totalCounter);
 
         // id dimension
-        CounterMetricFamily idCounter = new CounterMetricFamily(metricName + "&group=id",
-                "The metrics of inlong datastream.", this.dimensionKeys);
+        List<String> dimensionIdKeys = new ArrayList<>();
+        dimensionIdKeys.add(DEFAULT_DIMENSION_LABEL);
+        dimensionIdKeys.addAll(this.dimensionKeys);
+        CounterMetricFamily idCounter = new CounterMetricFamily(metricName + "_id",
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java:203]: Is it OK to change metric name?

**Review by luchunliang — APPROVED** [review]
OK

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java (hunk 3 above)
Comment (Keylchen) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java:203]: Special character will cause `expected text error`

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java (hunk 1 above)
Comment (Keylchen) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/metrics/prometheus/PrometheusMetricListener.java:202]: When prometheus server collect data by dimensionIdKeys, dimensionKeys will store keys  by snapshot interface.

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**